core/store/src/trie/shard_tries.rs

```diff
@@ -73,10 +73,12 @@ impl ShardTries {
                     shards[shard_id as usize].push((hash, Some(value.clone())));
                 }
                 DBOp::Insert { col, .. } if *col == DBCol::ColState => unreachable!(),
-                DBOp::Delete { col, key } if *col == DBCol::ColState => {
+                DBOp::Delete { col, .. } if *col == DBCol::ColState => unreachable!(),
+                DBOp::DeleteAll { col } if *col == DBCol::ColState => {
                     // Delete is possible in reset_data_pre_state_sync
-                    let (shard_id, hash) = TrieCachingStorage::get_shard_id_and_hash_from_key(key)?;
-                    shards[shard_id as usize].push((hash, None));
+                    for shard_id in 0..self.caches.len() {
+                        self.caches[shard_id].clear();
+                    }
```

Comment:
Are there any tests checking cache clearing / discrepancy between cache and column?

Reply:
A discrepancy manifests itself as `StorageInconsistentState`: the invariant is that if a trie node is in cache+storage, then its children have to be there too, and forgetting to clear cache can break it.
At the moment ViewClient has its own cache and allows the error, and regular Client panics if it happens.

Reply:
what was the reason that we allow the error in view client (instead of cleaning the cache)?

Reply:
Tracked by #3742. Two reasons:
1. Invalidating the cache requires taking a lock and we don't want ViewClient to block Client 
2. The error is theoretically possible in ViewClient because it doesn't read from snapshots. Client is single threaded and the only one who can write to storage, so it doesn't have that problem. If we ever decide to use snapshots we can bring back #3092.